common/txmgr/txmgr.go

```diff
@@ -56,6 +58,34 @@ type TxCandidate struct {
 	Value *big.Int
 }
 
+// SendTransaction sends a transaction to the chain.
+// If the transaction is successful, the receipt is returned, otherwise this process will be blocked.
+func (m *SimpleTxManager) SendTransaction(ctx context.Context, input []byte, to *common.Address, gasLimit uint64) (*types.Receipt, error) {
+	txCandidate := TxCandidate{
+		TxData:   input,
+		To:       to,
+		GasLimit: gasLimit,
+		Value:    big.NewInt(0),
+	}
+
+	receipt, err := m.Send(ctx, txCandidate)
+	if err != nil {
+		return nil, fmt.Errorf("send transaction: %w", err)
+	}
+
+	if receipt.Status != types.ReceiptStatusSuccessful {
```

Comment:
should not be handled in this function, return the receipt and let the invoker perform receipt status check.